### ISSUE TITLE
DR-> Unwanted validation error shows on the Find a protected area page

### ISSUE THREAD

**Issue by manuji** [bcgov/parks-data-register#328]
Scenario:
* Navigate to the DR on test environment: https://test-data.bcparks.ca/admin/protected-areas
* Login and navigate to the Find a protected area page
* Click on the text field to enter the search term and then click on the option to cancel
* Check the behavior. 

Issue:
* When you just click on the text field and then click on the "x" button, it shows validation errors for a moment. Refer to the video attached. 

Expected:
* Validation errors should not fire up unnecessarily unless the user tries to search without adding a search keyword. 

https://github.com/bcgov/parks-data-register/assets/65190263/1104692a-af38-43fe-87cf-c61458d0fef5

**Comment by cameronpettit** [comment]
Looks to be an issue with `ngds-toolkit` when using the inline clear button to clear an already empty field. I will see if I can fix that.

**Comment by cameronpettit** [comment]
@manuji @JLWade can confirm this issue arises when the following happens:

- The clear button is clicked on a field that is required to be populated when there is nothing in the field. 

This is due to Angular's order of operations. When clicking outside the field, the field loses focus ('blurs'). Since Angular is constantly polling the validity of the field, it knows the field will be invalid when clicked, but theres no logic that can be injected before the blur event fires so it will show as invalid until the mouse button is lifted.

I spent a lot of time trying to get ahead of the `blur` event and there doesnt appear to be any way to do it. I have attached an Angular issue for reference. 

I created some workarounds. We can either:

- hide the clear 'X' until it is safe to show it without the improper styling  (the field is `dirty`), or
- hide the invalid styling altogether (we can still check if the field is invalid, I've just updated the Forms Engine to be able to hide the invalid styling if necessary)
- Leave it as is, it is unlikely that it will be encountered often.
- 
Let me know what you want to do.

https://github.com/angular/angular/issues/7113
